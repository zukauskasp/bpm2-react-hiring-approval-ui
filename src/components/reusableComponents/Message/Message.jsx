import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './Message.scss';
import Icon from "../Icon/Icon";

const Message = ({ type, text, onClick }) => {

    const [icon, setIcon] = useState("");

    useEffect(() => {
        if (type !== 'default') setIcon(returnIconByName(type));
    }, [type]);

    return (
        <div className={`notification-wrapper ${type}-message`}>
            {icon ? <Icon icon={icon} color="white" className={`icon grid-item-first`} /> : null}
            <span className={`grid-item-second`}>{text}</span>
            <Icon icon="remove" color="white" onClick={onClick} className={`icon icon-remove grid-item-third `} />
        </div>
    );
}


export const returnIconByName = (messageType) => {
    let iconName;
    switch (messageType) {
        case "error":
            iconName = 'removeInCircle'
            break;
        case "warning":
            iconName = 'alert';
            break;
        case "success":
            iconName = 'check'
            break;
        case "info":
            iconName = "info"
            break;
        default:
            return "";
    }

    return iconName;

}

Message.propTypes = {
    type: PropTypes.oneOf(['error', 'warning', 'success', 'info', 'default']),
    text: PropTypes.string,
    onClick: PropTypes.func,
}

Message.defaultProps = {
    type: "default",
    text: ""
}

export default Message;