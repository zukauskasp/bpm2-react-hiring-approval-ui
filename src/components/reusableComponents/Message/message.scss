.notification-wrapper {
    height: 92px;
    width: 100%;
    color: white;
    font-size: 22px;
    line-height: 92px;
    font-family: DanskeHuman;
    font-weight: 500;
    font-style: italic;
    display: inline-grid;

    .icon {
        height: 1em;
        overflow: visible;
        vertical-align: -0.2em;
        width: 1em;
        margin: auto;
    }

    .icon-remove {
        cursor: pointer;
    }

    .grid-item-first {
        grid-column-start: 1;
        grid-column-end: 1;
    }

    .grid-item-second {
        grid-column-start: 2;
        grid-column-end: 10;
    }

    .grid-item-third {
        grid-column-start: 12;
        grid-column-end: 12;
    }
}

.error-message {
    background: #E65A6D;
}

.warning-message {
    background: #FBB273;
}

.success-message {
    background: #09B89D;
}

.info-message {
    background: #6DBACE;
}

.default-message {
    background: #003755;

}