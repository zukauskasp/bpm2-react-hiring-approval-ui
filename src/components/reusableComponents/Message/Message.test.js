import React from 'react';
import { shallow, mount } from 'enzyme';
import Message, { returnIconByName } from './Message';

describe('Message', () => {

    const onClick = (event) => event;
    let wrapper;
    const props = {
        type: "success",
        text: "This is text",
        onClick: onClick
    }

    beforeEach(() => {
        wrapper = shallow(<Message {...props} />);
    })

    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        expect(mount(<Message {...props} />).debug()).toMatchSnapshot();
    });

    it('expect to render Message', (() => {
        expect(wrapper.length).toEqual(1);
    }));

    it('expect to render correct props', () => {
        expect(mount(<Message {...props} />).props()).toEqual(props);
    });

    it('expect to have correct classes', (() => {
        const errorMessageWrapper = shallow(<Message type="error" />);

        expect(wrapper.prop('className')).toEqual('notification-wrapper success-message');
        expect(wrapper.find('span').prop('className')).toEqual('grid-item-second');
        expect(wrapper.find('Icon').prop('className').indexOf('icon') > -1).toEqual(true);
        expect(errorMessageWrapper.prop('className')).toEqual('notification-wrapper error-message');
    }));

    it('expect to render default props', () => {
        const defaultProps = {
            type: "default",
            text: "",
        };

        expect(mount(<Message />).props()).toEqual(defaultProps);
    });

    it('expect to simulate event', () => {
        const mockCallBack = jest.fn();

        shallow(<Message onClick={mockCallBack} />).find("Icon").simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('expect returnIconByName to return correct icon name', () => {
        let icon = returnIconByName(props.type);

        expect(icon).toEqual('check');
        expect(icon === "wrong").toEqual(false);
    });

}); 