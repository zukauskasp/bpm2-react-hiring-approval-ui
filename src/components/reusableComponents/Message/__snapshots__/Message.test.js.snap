// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message expect to match latest snapshot 1`] = `
"<Message type=\\"success\\" text=\\"This is text\\" onClick={[Function: onClick]}>
  <div className=\\"notification-wrapper success-message\\">
    <Icon icon=\\"check\\" color=\\"white\\" className=\\"icon grid-item-first\\">
      <img alt=\\"\\" src=\\"/assets/icons/check/white.svg\\" className=\\"icon grid-item-first\\" />
    </Icon>
    <span className=\\"grid-item-second\\">
      This is text
    </span>
    <Icon icon=\\"remove\\" color=\\"white\\" onClick={[Function: onClick]} className=\\"icon icon-remove grid-item-third \\">
      <img alt=\\"\\" src=\\"/assets/icons/remove/white.svg\\" onClick={[Function: onClick]} className=\\"icon icon-remove grid-item-third \\" />
    </Icon>
  </div>
</Message>"
`;
